ipfs-pinning-service.yaml

```diff
@@ -141,66 +193,134 @@ paths:
 
 components:
   schemas:
-    Error:
+
+    PinStatus:
+      description: Pin object with Status
       type: object
       properties:
-        code:
-          type: integer
-        message:
+        id:
+          description: CID of Pin object that can be used for status checks of ongoing pinning
           type: string
+        status:
+          $ref: '#/components/schemas/Status'
+        pin:
+          $ref: '#/components/schemas/Pin'
 
-    Cid:
+    Pin:
+      description: Pin object
       type: object
       properties:
-        /:
+        cid:
+          description: CID to be pinned
           type: string
+        depth:
+          description: Defines how deep the DAG should be pinned (-1 recursive, 0 direct)
+          type: integer
+          format: int32
+          default: -1
+          minimum: -1
+        meta:
+          $ref: '#/components/schemas/Meta'
+      required:
+      - cid
+
+    Status:
+      description: Status in which pin object can exist at a pinning service
+      type: string
+      enum:
+        - pinned
+        - pinning  # in-progress
+        - removing # in-progress
+        - expired  # still pinned for some time, but run out of funds and is not provided to the IPFS network
 
-    Metadata:
+    Meta:
+      description: Optional metadata for Pin object
       type: object
       additionalProperties:
         type: string
 
-    ReplicationFactor:
-      type: integer
-      minimum: -1
-
-    PinStatus:
+    Error:
+      description: Base Error object
       type: object
       properties:
-        peername:
-          type: string
-        peer:
-          type: string
-        error:
+        code:
+          type: integer
+        message:
           type: string
-        pin:
-          $ref: '#/components/schemas/Pin'
-        status:
+
+  parameters:
+
+    skip:
+      description: number of items to skip
+      name: skip
+      in: query
+      required: false
+      schema:
+        type: integer
+        format: int32
+
+    limit:
+      description: max records to return
+      name: limit
+      in: query
+      required: false
+      schema:
+        type: integer
+        format: int32
```

Comment:
Should we add the minimum(1), maximum(1000?) and default(max?) here? While services could vary I think it would be good to just lock this in for simplicity.